sound/soc/sof/core.c

```diff
@@ -249,6 +250,12 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 	if (plat_data->sof_probe_complete)
 		plat_data->sof_probe_complete(sdev->dev);
 
+	ret = sof_register_clients(sdev);
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: failed to register clients %d\n", ret);
+		goto fw_trace_err;
```

Comment:
I think I'm also missing a call to `snd_sof_machine_unregister()` if the client registration fail, we are after `snd_sof_machine_register()` at this point.